toolip/utils/logs.py

```diff
@@ -0,0 +1,29 @@
+import logging
+
+from loguru import logger
+
+
+class InterceptHandler(logging.Handler):
+    def emit(self, record):
+        # Get corresponding Loguru level if it exists
+        try:
+            level = logger.level(record.levelname).name
+        except ValueError:
+            level = record.levelno
+
+        # Find caller from where originated the logged message
+        frame, depth = logging.currentframe(), 2
+        while frame.f_code.co_filename == logging.__file__:
+            frame = frame.f_back
+            depth += 1
+
+        logger.opt(depth=depth, exception=record.exc_info).log(level, record.getMessage())
+
+
+def setup_logging(logfilename: str, rotation: str, retention: str) -> None:
+    """ Catch and redirect the loggers already defined in uvicorn """
+    logging.getLogger('uvicorn.access').handlers = [InterceptHandler()]
+    logging.getLogger('fastapi').handlers = [InterceptHandler()]
```

Comment:
Can you test this function in a script that doesn't run uvicorn or fastapi??
I'm worried that it will crash in those cases.

Reply:
@hillairet Tested this one and seems to work fine without uvicorn and fastapi. I just ran a script in a virtualenv that had nothing related to uvicorn and fastapi.

Reply:
Sweet!